spark/src/main/scala/ai/chronon/spark/GroupBy.scala

```diff
@@ -515,8 +515,8 @@ object GroupBy {
     // Generate mutation Df if required, align the columns with inputDf so no additional schema is needed by aggregator.
     val mutationSources = groupByConf.sources.toScala.filter { _.isSetEntities }
     val mutationsColumnOrder = inputDf.columns ++ Constants.MutationFields.map(_.name)
-    val mutationDf =
-      if (mutationScan && groupByConf.inferredAccuracy == api.Accuracy.TEMPORAL && mutationSources.nonEmpty) {
+    def mutationDfFn(): DataFrame = {
```

Comment:
This makes mutationDf eval lazy.

So that it is only touched when temporal entities code is run

Reply:
nice!